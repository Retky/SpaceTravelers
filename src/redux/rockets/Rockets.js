import fetchRockets from './api';

const FETCHROCKETS = 'FETCHROCKETS';
const RESERVEROCKET = 'RESERVEROCKET';
const UNRESERVEROCKET = 'UNRESERVEROCKET';
const initialState = [];

export const getRockets = () => async (dispatch) => {
  const newState = await fetchRockets();
  dispatch({
    type: FETCHROCKETS,
    newState,
  });
};

export const reserveRocket = (id) => ({
  type: RESERVEROCKET,
  id,
});

export const unreserveRocket = (id) => ({
  type: UNRESERVEROCKET,
  id,
});

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCHROCKETS:
      return action.newState;
    case RESERVEROCKET: {
      const valState = state.map((rocket) => {
        if (rocket.rocket_id !== action.id) {
          return rocket;
        }
        return { ...rocket, reserved: true };
      });
      console.log(valState);
      return valState;
    }
    case UNRESERVEROCKET: {
      const valState = state.map((rocket) => {
        if (rocket.rocket_id !== action.id) {
          return rocket;
        }
        return { ...rocket, reserved: false };
      });
      console.log(valState);
      return valState;
    }
    default:
      return state;
  }
};

export default reducer;
