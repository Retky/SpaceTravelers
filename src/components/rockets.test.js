import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import Rockets from './rockets';

describe('Rockets component', () => {
  test('component match with the snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
