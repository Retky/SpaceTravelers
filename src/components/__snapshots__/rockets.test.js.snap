// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component component match with the snapshot 1`] = `
<section>
  <ul
    className="rocketsList"
  />
</section>
`;
